hybrasyl/Objects/Creature.cs

```diff
@@ -557,23 +978,14 @@ public virtual void Motion(byte motion, short speed)
             }
         }
 
-        //public virtual bool AddItem(ItemObject item, bool updateWeight = true) { return false; }
-        //public virtual bool AddItem(ItemObject item, int slot, bool updateWeight = true) { return false; }
-        //public virtual bool RemoveItem(int slot, bool updateWeight = true) { return false; }
-        //public virtual bool RemoveItem(int slot, int count, bool updateWeight = true) { return false; }
-        //public virtual bool AddEquipment(ItemObject item) { return false; }
-        //public virtual bool AddEquipment(ItemObject item, byte slot, bool sendUpdate = true) { return false; }
-        //public virtual bool RemoveEquipment(byte slot) { return false; }
-
-        public virtual void Heal(double heal, Creature target = null)
+        public virtual void Heal(double heal, Creature source = null)
         {
-            if (target == null) target = this;
-            if (target.AbsoluteImmortal || target.PhysicalImmortal) return;
-            if (target.Hp == target.MaximumHp) return;
+            if (AbsoluteImmortal || PhysicalImmortal) return;
+            if (Hp == MaximumHp) return;
 
-            target.Hp = heal > uint.MaxValue ? target.MaximumHp : Math.Min(target.MaximumHp, (uint)(target.Hp + heal));
-            SendDamageUpdate(target);
-            if (target is User) { target.UpdateAttributes(StatUpdateFlags.Current); }
+            Hp = heal > uint.MaxValue ? MaximumHp : Math.Min(MaximumHp, (uint)(Hp + heal));
+            SendDamageUpdate(this);
```

Comment:
this would work for self heal.. not healing a target

Reply:
`Heal` heals `this`. Healing another creature should be done via `othercreatureobj.Heal(3)`, not `this.Heal(3, othercreatureobj)`. Instances of healing elsewhere all use this pattern now and pass `source` as the healer object for future use.